wallet/src/state.rs

```diff
@@ -85,7 +88,7 @@ impl AsRef<Note> for UnspentNote<'_> {
 }
 
 impl ClientState {
-    pub fn new(wallet: Wallet) -> Self {
+    pub fn new(wallet: Wallet, chain_params: Option<ChainParams>) -> Self {
```

Comment:
The `ClientState` is only ever constructed with `chain_params = None`, so maybe it would be simpler to skip passing it as an argument, set it to `None` by default, and use the existing logic to set it if it doesn't exist?